test/test_mount.rs

```diff
@@ -207,15 +207,22 @@ macro_rules! run_tests {
 
 #[cfg(target_os = "linux")]
 fn main() {
-    use test_mount::{setup_namespaces, test_mount_tmpfs_without_flags_allows_rwx,
-                     test_mount_rdonly_disallows_write, test_mount_noexec_disallows_exec,
-                     test_mount_bind};
-    setup_namespaces();
-
-    run_tests!(test_mount_tmpfs_without_flags_allows_rwx,
-               test_mount_rdonly_disallows_write,
-               test_mount_noexec_disallows_exec,
-               test_mount_bind);
+    use std::env;
+
+    if env::var("DOCKER_ENVIRONMENT").is_ok() {
```

Comment:
Ok I just sent #327 to allow it to fail if the reason is `unshare` not supported.

@posborne you should be able to rebase on top of that once it's accepted, and drop changes in this file.